test/functional/wallet_conflicts.py

```diff
@@ -28,6 +29,20 @@ def get_utxo_of_value(self, from_tx_id, search_value):
         return next(tx_out["vout"] for tx_out in self.nodes[0].gettransaction(from_tx_id)["details"] if tx_out["amount"] == Decimal(f"{search_value}"))
 
     def run_test(self):
+        """
+        The following tests check the behavior of the wallet when
```

Comment:
In commit "test: Add tests for wallet mempool conflicts" (c891a65067f57ba87e443d0cdc4f7e6b434c436f)

Ignore this suggestion if it requires too much work, but IMO it would be really helpful for review if as much of the new test code as possible was added in the first commit before the C++ code changes, and the tests were updated alongside the code changes in subsequent commits.

Independently of this PR, the fact that a commit like 2c867784b2a719c9e701e19e389c3adfa7d7faae can make a number of behavior changes and not need any significant test updates shows that we are lacking a lot of test coverage for these code paths presently. So it would be very reassuring to add the test coverage before making the changes and see the changes having the expected effects on tests, instead of making the changes blindly and adding the test coverage later.

Reply:
I have made the first commit the commit with tests, and they are updated to reflect the changes in the following commits.